model/helics-application.cc

```diff
@@ -349,7 +350,8 @@ HelicsApplication::Send (std::string dest, std::unique_ptr<helics::Message> mess
   // so that tags added to the packet can be sent as well
   m_txTrace (p);
   
-  int delay_ns = (int) (m_rand_delay_ns->GetValue (m_jitterMinNs,m_jitterMaxNs) + 0.5);
+  int64_t delay_ns = Time::FromDouble (time, Time::S).GetNanoSeconds() - Simulator::Now ().GetNanoSeconds () + (int64_t) (m_rand_delay_ns->GetValue (m_jitterMinNs,m_jitterMaxNs) + 0.5);
+  // int delay_ns = (int) (m_rand_delay_ns->GetValue (m_jitterMinNs,m_jitterMaxNs) + 0.5);
```

Comment:
Breaking up the long line a little bit and adding a comment to remind why this is needed (an ns-3 ScheduleAt function that takes an absolute time would be nice).
```suggestion
  int64_t jitter_ns = (int) (m_rand_delay_ns->GetValue (m_jitterMinNs,m_jitterMaxNs) + 0.5);
  
  // Calculate the delay between when the event happens relative to current ns-3 simulation time
  int64_t delay_ns = Time::FromDouble (time, Time::S).GetNanoSeconds() - Simulator::Now ().GetNanoSeconds () + jitter_ns;
```

Reply:
Changes have been done and tested.